src/diffusers/models/transformers/transformer_bria_fibo.py

```diff
@@ -0,0 +1,446 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ...loaders import FromOriginalModelMixin, PeftAdapterMixin
+from ...models.attention_processor import Attention
+from ...models.embeddings import TimestepEmbedding, get_timestep_embedding
+from ...models.modeling_outputs import Transformer2DModelOutput
+from ...models.modeling_utils import ModelMixin
+from ...models.normalization import AdaLayerNormContinuous, AdaLayerNormZeroSingle
+from ...models.transformers.transformer_bria import BriaAttnProcessor
+from ...models.transformers.transformer_flux import FluxTransformerBlock
+from ...utils import (
+    USE_PEFT_BACKEND,
+    logging,
+    scale_lora_layers,
+    unscale_lora_layers,
+)
+from ...utils.torch_utils import maybe_allow_in_graph
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+def get_1d_rotary_pos_embed(
```

Comment:
Is it different from
https://github.com/huggingface/diffusers/blob/500b9cf184d9a6aa950ccbb9a4d967b95877196d/src/diffusers/models/embeddings.py#L1120 ?

Can we repurpose it?